renderer/renderer.go

```diff
@@ -64,14 +64,60 @@ func combinedFuncMap(funcs ...funcMap) template.FuncMap {
 	return m
 }
 
-func createOutFile(outputPath string, defaultFileName string) (*os.File, error) {
+type renderObject struct {
+	fileName    string
+	expectedDir bool
+	gvd         []types.GroupVersionDetails
+}
+
+func renderTemplate(tmpl *template.Template, conf *config.Config, fileFormat string, gvd []types.GroupVersionDetails) error {
+	renderObjects := make([]renderObject, 0)
+	switch conf.OutputMode {
+	case config.OutputModeSingle:
+		object := renderObject{
+			fileName: fmt.Sprintf("%s.%s", "out", fileFormat),
+			gvd:      gvd,
+		}
+		renderObjects = append(renderObjects, object)
+	case config.OutputModeGroup:
+		for _, item := range gvd {
+			object := renderObject{
+				fileName:    fmt.Sprintf("%s.%s", item.Group, fileFormat),
+				gvd:         []types.GroupVersionDetails{item},
+				expectedDir: true,
+			}
+			renderObjects = append(renderObjects, object)
+		}
+	}
+
+	return render(tmpl, conf.OutputPath, renderObjects)
+}
+
+func render(tmpl *template.Template, outputPath string, objects []renderObject) error {
+	for _, object := range objects {
+		file, err := createOutFile(outputPath, object.expectedDir, object.fileName)
+		defer file.Close()
+		if err != nil {
+			return err
+		}
+
+		if err := tmpl.ExecuteTemplate(file, mainTemplate, object.gvd); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
The intermediate struct doesn't bring a lot imo. I would change `renderTemplate` like this:
```suggestion
func renderTemplate(tmpl *template.Template, conf *config.Config, fileExtension string, gvds []types.GroupVersionDetails) error {
	switch conf.OutputMode {
	case config.OutputModeSingle:
		file, err := createOutFile(conf.OutputPath, false, fmt.Sprintf("%s.%s", "api-docs", fileExtension))
		if err != nil {
			return err
		}
		defer file.Close()
		return tmpl.ExecuteTemplate(file, mainTemplate, gvds)

	case config.OutputModeGroup:
		for _, gvd := range gvds {
			file, err := createOutFile(conf.OutputPath, true, fmt.Sprintf("%s.%s", gvd.Group, fileExtension))
			if err != nil {
				return err
			}
			defer file.Close()
			if err := tmpl.ExecuteTemplate(file, mainTemplate, []types.GroupVersionDetails{gvd}); err != nil {
				return nil
			}
		}
	}
	return nil
}
```